### ISSUE TITLE
Add a Flutter module to an Android library project

### ISSUE THREAD

**Issue by jronceroscb** [flutter/flutter#95345]
Hello there.

For the past year, I've been able to build an Android library with an embedded Flutter module inside. The way I've done this is following the "Option B - Depend on the module’s source code" in the [Integrate a Flutter module](https://docs.flutter.dev/development/add-to-app/android/project-setup#option-b---depend-on-the-modules-source-code) guide.

I had to make some tweaks to my implementation and also modify the flutter.gradle file inside flutter_tools folder of the flutter SDK so it can compile inside a library type project. This workaround is really not the way I want to proceed since I guess it's not how it should be done.

Before all of that, I tried to use "Option A - Depend on the Android Archive (AAR)", but I had problems when exporting my own Android library with the Flutter module inside as an aar, since I couldn't make a proper fat aar.

TLDR; Is there a way to integrate a Flutter module into an Android Library project and not into an Android App project? How should I proceed?

**Comment by maheshmnj** [comment]
Hi @jronceroscb, The possible ways to add flutter module to an existing android  app has been documented in [add to app docs](https://docs.flutter.dev/development/add-to-app/android/project-setup). if you need further help or have any questions I would encourage you to please use one of the supported channels at https://flutter.dev/community. You may also get some help if you post on of the discussion forums like StackOverflow or reddit or you could reach out to flutter team members on discord.

Closing as this issue doesn't seem to describe a bug or a feature request.

**Comment by jronceroscb** [comment]
As I said already, that documentation is to add the flutter module into an **app,** not a **library**. Hence my qustion.

**Comment by maheshmnj** [comment]
Apologies missed that info, Reopening based on the above comment.

**Comment by maheshmnj** [comment]
Leaving this issue open for further insights from the team.

**Comment by jronceroscb** [comment]
Thanks @maheshmnj for reopening. I will add more information of my current implementation in the first comment.

**Comment by jronceroscb** [comment]
Added more information regarding my current implementation

**Comment by pjmdr11** [comment]
Hi! were you able to find any other solution for integrating the flutter module to android libraries?

**Comment by jorgeroncero** [comment]
@pjmdr11 Hi! No, I didn't find any other solution. I am still using the mentioned workaround (modifying flutter.gradle)

**Comment by devAseem-Piie** [comment]
What Gradle version have you used for your solutions @jorgeroncero I am getting error while adding your fix.

`org.gradle.api.InvalidUserCodeException: Cannot run Project.afterEvaluate(Closure) when the project is already evaluated.`

**Comment by ddthanh198** [comment]
How can I build a fat AAR file from flutter module which include GeneratedPluginRegistrant and all plugin files?

**Comment by bugrahancamlibel** [comment]
Is there anything new in here? I am trying to publish my Flutter module to GitHub Packages. As I see that it is not supported by Flutter. So I tried to do a workaround like adding Flutter module to an AAR so I can publish it. Now I'm stuck in this step. Any updates in here?

**Comment by kuba-asanov** [comment]
Is there any updates?
I have tried with aar and also with project(“:flutter”) but any results, after publication to Maven Cetral my library can’t find libflutter.so. Could anyone published android library with depends from flutter?
